mmv1/templates/terraform/pre_delete/firestore_database.go.erb

```diff
@@ -0,0 +1,26 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2023 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+
+// start of customized code
+deletionPolicy := d.Get("deletion_policy");
+
+if deletionPolicy == "ABANDON" {
+	return nil
+}
+
+if d.Get("delete_protection_state").(string) == "DELETE_PROTECTION_ENABLED" {
```

Comment:
I think this is not required (since the API would catch this, and the user is already about to make the API call anyway) - let me know if you think it's useful to have here, though